LOAD CSV warnings

henriknyman opened the pull request:
- Warn when a LOAD CSV is followed by an Eager 
- Warn when LOAD CSV plus MATCH/MERGE on large non-indexed label

henriknyman committed:
Generate warning on LOAD CSV with Eager operator

When a LOAD CSV is followed by an Eager operator that forces all data
to be materialized in memory a warning about potential problems
with memory consumption is generated.

pontusmelke commented on community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/ExecutionPlanBuilder.scala:
```diff
@@ -142,13 +143,40 @@ class ExecutionPlanBuilder(graph: GraphDatabaseService, statsDivergenceThreshold
+  }
+
+
+  def checkForNotifications(pipe: Pipe): Seq[InternalNotification] = {
```
private

jexp commented:
I can confirm that it generates the warning.

pontusmelke committed:
Warnings for LOAD CSV + large non-indexed label

jexp commented:
Probably split into two PRs 
The first one was already ok and could have been merged :)

pontusmelke commented:
@jexp Yeah we should probably, the second one depends on the first though

pontusmelke committed:
Clean up and added check for merge

- moved checks into separate files
- wrote unit tests for checkers
- added warnings for merge
